pkg/resourcemanager/webhook/endpointslicehints/handler_test.go

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) 2023 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package endpointslicehints_test
+
+import (
+	"context"
+
+	"github.com/gardener/gardener/pkg/logger"
+	. "github.com/gardener/gardener/pkg/resourcemanager/webhook/endpointslicehints"
+
+	"github.com/go-logr/logr"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	discoveryv1 "k8s.io/api/discovery/v1"
+	"k8s.io/utils/pointer"
+	logzap "sigs.k8s.io/controller-runtime/pkg/log/zap"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+var _ = Describe("Handler", func() {
```

Comment:
What do you think about dropping this unit test? The integration test already covers the same scenarios.

Reply:
We should rather agree on this and potentially formulate it as rule like "Do not add unit tests for webhooks, integration tests are sufficient.". Currently I see that the system-components-config, high-availability-config, extension-validation webhooks don't have unit tests, other have.

Reply:
We can have this as "rule of thumb", but it still depends on the concrete reconciler/webhook handler. Integration tests are heavyweight and should rather focus on testing the integration. If the reconciler has many responsibilities and branches, then unit tests might be more appropriate (since they run faster/are more lightweight). In this concrete case, unit tests don't make much sense since the logic is small and straight-forward.